### PR TITLE
[GraphBolt] Edge feature fetch fix.

### PR DESCRIPTION
**Pull request by mfbalin**
## Description
The modified test fails. In this PR, I plan to fix it by always enabling return_eids=True when sampling.

EDIT: Looks like edge feature fetch was broken all this time. Our tests do actually cover it but assertions were not written to check for it.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by mfbalin** [comment]
The distributed tests run successfully on my local @Rhett-Ying

File: python/dgl/graphbolt/feature_fetcher.py [hunk 1]
```diff
@@ -215,8 +215,6 @@ def wait(self):
         if self.edge_feature_keys and num_layers > 0:
             for i in range(num_layers):
                 original_edge_ids = data.edge_ids(i)
-                if original_edge_ids is None:
```
Comment (frozenbugs) [inline python/dgl/graphbolt/feature_fetcher.py:218]: why is necessary to remove this check?

Comment (mfbalin) [reply python/dgl/graphbolt/feature_fetcher.py:218]: If features are available, feature fetcher silently runs and fetches no features due to this line. This was when return eids is False. Do you think we should remove this line?

Comment (mfbalin) [reply python/dgl/graphbolt/feature_fetcher.py:218]: Yes, I expect it to be never None.

**Comment by mfbalin** [comment]
> I am confused by this PR, why always enabling return_eids=True when sampling?

https://github.com/dmlc/dgl/pull/7589#discussion_r1696826258

File: python/dgl/graphbolt/feature_fetcher.py (hunk 1 above)
Comment (mfbalin) [reply python/dgl/graphbolt/feature_fetcher.py:218]: And the tests were broken, not testing edge feature fetch at all.

**Pull request merged**